pkg/store/bucket.go

```diff
@@ -1261,7 +1302,25 @@ func (s *BucketStore) LabelValues(ctx context.Context, req *storepb.LabelValuesR
 		}
 	}
 
+	// If we have series matchers, add <labelName> != "" matcher, to only select series that have given label name.
+	if len(reqSeriesMatchers) > 0 {
```

Comment:
What if `reqSeriesMatchers` already contains this unequal matcher? @pstibrany @pracucci

Reply:
End result will be correct, but there is an optimization we can do (preferably in `blockSeries` to be more general).

Reply:
Make sense. Thank you.